File: BookNestBackend/BookNest.Services/Database/Entities/City.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookNest.Services.Database.Entities
{
    public class City
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        public int CountryId { get; set; }

        [ForeignKey(nameof(CountryId))]
        public Country Country { get; set; } = null!;

        public ICollection<User> Users { get; set; } = new List<User>();
        public ICollection<Shipping> Shippings { get; set; } = new List<Shipping>();
        public ICollection<Event> Events { get; set; } = new List<Event>();
    }
}

File: BookNestFrontend/booknest_mobile/lib/models/city.dart
class City {
  final int id;
  final String name;
  final int countryId;
  final String countryName;

  City({
    required this.id,
    required this.name,
    required this.countryId,
    required this.countryName,
  });

  factory City.fromJson(Map<String, dynamic> json) {
    return City(
      id: json['id'],
      name: json['name'] ?? '',
      countryId: json['countryId'] ?? 0,
      countryName: json['countryName'] ?? '',
    );
  }
}